src/main/java/org/schedulesdirect/api/ZipEpgClient.java

```diff
@@ -286,11 +302,17 @@ protected Program fetchProgram(final String progId) throws IOException {
 		Program p = progCache.get(progId);
 		if(p == null) {
 			Path path = vfs.getPath(String.format("programs/%s.txt", scrubFileName(progId)));
+			
+			if(!Files.exists(path)) {
```

Comment:
Only check the seriesInfo dir if and only if the progId starts with "SH" otherwise we know it can't ever exist in there so let's save the I/O overhead of searching inside the zip fs when we know it can't ever be in there.